amida-tech/indaba-client
# Bug/inba 917 jf fix for login

#### What does this PR do?
This is a refactor of the login page. There was some bloat and some features we didn't really want or need in there. The PR was trying to remove the "click twice" issue where sometimes the login would stall. I think it succeeds.

#### Related JIRA tickets:
INBA-917.

#### How should this be manually tested?
This is a total rewrite so we have to check all functionality for login:
1) As soon as you load the page, remove any emails saved to the fields and reload. 
2) Upon reload, with blank fields, immediately hit "Sign In". The fields should turn red and tell the user that they are required.
3) Refresh the page.
4) Click on the "Email" field then click away. The field should turn red, and display "Email required"
5) Click on the "Password" field then click away. The field should turn red, and display "Password required" 
6) Fill in email and confirm that the red goes away around its field.
7) Fill in password and confirm that the red goes away around its field.
8) Fill in username and password. Click "Sign In". Confirm you sign in.
9) Log out. Repeat the login, but this time hit the "Enter" key. Confirm you sign in.
10) Logout. With a blank email field, click on "Forgot password?" Confirm no toast is displayed. Instead, the user should be told to fill in their email in the field above for the modal.
11) Fill in the email and click on "Forgot Password?" Try canceling.
12) Click on "Forgot Password?" and confirm a reset. Confirm the message about sending an email is set on the screen and not by a toast. Confirm you receive the email.
13) Click the "Remember me?" checkbox and sign in. Make sure your amida-auth-microservice has `.env` setting `AUTH_SERVICE_REFRESH_TOKEN_ENABLED=true` or this will not work (it may have been just `REFRESH_TOKEN_ENABLED` before). Then walk away for an hour or so and check back. You should still be logged in. 

Then, after receiving the email, check the following for the reset password screen:
1) Upon load, with blank fields, immediately hit "Reset password". The fields should turn red and tell the user that they are required.
2) Refresh the page.
4) Click on the "New Password" field then click away. The field should turn red, and display "Password required"
5) Click on the "Confirm Password" field then click away. The field should turn red, and display "Password confirmation required" 
6) Fill in password and confirm that the red goes away around its field.
7) Fill in confirm password and confirm that the red goes away around its field.
8) Fill in two passwords that don't match and hit "Reset Password" to receive warning about mismatched passwords.
9) Fill two passwords that match. Turn OFF amida-auth-microservice. Submit. Confirm you receive a failed message.
10) Turn on amida-auth-microservice. Submit with two matching passwords. Confirm it goes through and you are redirected. No toasts should appear.

I actually cannot remember if that last point still works or not. If not, make it another ticket.

Reedy, please check IE. Riley, feel free to weigh in on UI but understand I may draw the line if it's a lot of work. 

#### Background/Context

#### Screenshots (if appropriate):
